armi/reactor/blocks.py

```diff
@@ -1559,26 +1560,49 @@ class during axial expansion.
         """
         self.p.axialExpTargetComponent = targetComponent.name
 
-    def getPinCoordinates(self):
+    def getPinLocations(self) -> list[grids.IndexLocation]:
+        """Produce all the index locations for pins in the block.
+
+        Returns
+        -------
+        list[grids.IndexLocation]
+            Integer locations where pins can be found in the block.
+
+        Notes
+        -----
+        Only components with ``Flags.CLAD`` are considered to define a pin's location.
```

Comment:
I know this isn't part of your design, but maybe this method could take an optional `flag` parameter that defaults to `Flags.CLAD`. That way, if in the future someone wants to use some other flag to identify what they want to define as a "pin".

This request is scope creep, but it seems like a one-line change and a single unit test to make this more extensible in ways that you have convinced me are important.

Thoughts?

Reply:
Are you okay if I make a ticket and/or attach this to a discussion? While `getPinLocations(Flags.FUEL)` tracks, more generally trying to find locations of children seems useful. But, calling `getPinLocations(Flags.POISON)` feels weird.

Reply:
This can be a separate ticket, NBD.  It's a very minor idea, and won't hold up this PR either way.